### PR TITLE
Build on macOS with Rider

### PR DESCRIPTION
**Pull request by slang25**
Hey, congrats on the v4.0.0 release. I've put some love into getting this solution working well with Rider on macOS.

### CODE REVIEW

**Comment by martincostello** [comment]
Also should add this property to the new targets/props files so the build isn't treated as up-to-date if they're edited.

https://github.com/justeat/JustEat.StatsD/blob/0b6e5e0a3d5dbb856b7b25cca7f32b1c51f993aa/Directory.Build.props#L20

**Comment by martincostello** [comment]
Does this need updating for Linux too? There's [a warning](https://travis-ci.org/justeat/JustEat.StatsD/jobs/482550311#L577) in the build.

```sh
/home/travis/build/justeat/JustEat.StatsD/.dotnetcli/sdk/2.2.103/Microsoft.Common.CurrentVersion.targets(2110,5): warning MSB3245: Could not resolve this reference. Could not locate the assembly "System.Xml". Check to make sure the assembly exists on disk. If this reference is required by your code, you may get compilation errors. [/home/travis/build/justeat/JustEat.StatsD/src/JustEat.StatsD/JustEat.StatsD.csproj]
```

File: src/JustEat.StatsD/StatsDConfiguration.cs [hunk 1]
```diff
@@ -42,7 +42,7 @@ public class StatsDConfiguration
 
         /// <summary>
         /// Gets or sets the socket protocol to use, such as using either UDP or IP
-        /// sockets to transport stats. The default value is <see cref="SocketProtocol.Udp"/>.
+        /// sockets to transport stats. The default value is <see cref="JustEat.StatsD.SocketProtocol.Udp"/>.
```
[the remark this answers is not in the document]
Comment (slang25) [reply src/JustEat.StatsD/StatsDConfiguration.cs]: It does not, and I couldn't find any info about what is and isn't legal. You can reference members of the current class but values have to be fully qualified for Rider not to report it as an error :shrug:

**Comment by slang25** [comment]
> Does this need updating for Linux too? There's [a warning](https://travis-ci.org/justeat/JustEat.StatsD/jobs/482550311#L577) in the build.
> 
> ```shell
> /home/travis/build/justeat/JustEat.StatsD/.dotnetcli/sdk/2.2.103/Microsoft.Common.CurrentVersion.targets(2110,5): warning MSB3245: Could not resolve this reference. Could not locate the assembly "System.Xml". Check to make sure the assembly exists on disk. If this reference is required by your code, you may get compilation errors. [/home/travis/build/justeat/JustEat.StatsD/src/JustEat.StatsD/JustEat.StatsD.csproj]
> ```

It's not ideal, net45x reference assemblies include a file named `System.XML.dll` instead of `System.Xml.dll`, it should be possible to rename either the reference or the filename so that they line up, but I'm struggling. It's a legit warning, if we start to use a `System.Xml` type the build target will fail.

File: Directory.Build.props [hunk 2]
```diff
@@ -13,6 +13,7 @@
     <Authors>JUSTEAT_OSS</Authors>
     <Company>Just Eat</Company>
     <Copyright>Copyright (c) Just Eat 2015-$([System.DateTime]::Now.ToString(yyyy))</Copyright>
+    <MSBuildAllProjects>$(MSBuildAllProjects);$(MSBuildThisFileFullPath)</MSBuildAllProjects>
```
Comment (martincostello) [inline Directory.Build.props]: You put this in the wrong file - now there's two in this one 😄

File: src/JustEat.StatsD/JustEat.StatsD.csproj [hunk 3]
```diff
@@ -13,6 +13,7 @@
   <ItemGroup Condition=" '$(TargetFramework)' == 'net451' ">
     <Reference Include="System" />
     <Reference Include="System.Runtime" />
+    <Reference Include="System.Xml" />
```
Comment (martincostello) [inline src/JustEat.StatsD/JustEat.StatsD.csproj]: wat?

Comment (slang25) [reply src/JustEat.StatsD/JustEat.StatsD.csproj]: This is an old commit, I was testing using an actual `System.Xml` type with a workaround to see if I could prevent the potential issue, which I did in a later commit and reverted it cause I couldn't get it working.

File: Directory.Build.props [hunk 4]
```diff
@@ -1,5 +1,8 @@
 <Project>
  <Import Project="version.props" />
+  <PropertyGroup>
+    <MSBuildAllProjects>$(MSBuildAllProjects);$(MSBuildThisFileFullPath)</MSBuildAllProjects>
```
Comment (martincostello) [inline Directory.Build.props]: This one's still a duplicate 😉

**Comment by JosephWoodward** [comment]
I approve of this PR 👍 ❤️

**Pull request merged**